File: tags/Versión Marcas y Patentes Internacionales/Cliente/Cliente/Validaciones/CorreoElectronico.cs
using System.Windows.Controls;
using System.Text.RegularExpressions;

namespace Trascend.Bolet.Cliente.Validaciones
{
    public class CorreoElectronico : ValidationRule
    {
        private const string _expresionRegular = "\\w+([-+.']\\w+)*@\\w+([-.]\\w+)*\\.\\w+([-.]\\w+)*";

        public override ValidationResult Validate(object value, System.Globalization.CultureInfo cultureInfo)
        {
            string valor = (string) value;

            Regex regular = new Regex(_expresionRegular);

            if (!regular.IsMatch(valor) && !string.IsNullOrEmpty(valor))
            {
               return new ValidationResult(false, Recursos.MensajesValidaciones.CorreoElectronico);
            }
            else
                return ValidationResult.ValidResult;
        }
    }
}

File: trunk/src/Cliente/Cliente_Fac/Validaciones/CorreoElectronico.vb
Imports System.Windows.Controls
Imports System.Text.RegularExpressions

Namespace Validaciones
    Public Class CorreoElectronico
        Inherits ValidationRule
        Private Const _expresionRegular As String = "\w+([-+.']\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*"

        Public Overrides Function Validate(ByVal value As Object, ByVal cultureInfo As System.Globalization.CultureInfo) As ValidationResult
            Dim valor As String = DirectCast(value, String)

            Dim regular As New Regex(_expresionRegular)

            If Not regular.IsMatch(valor) AndAlso Not String.IsNullOrEmpty(valor) Then
                Return New ValidationResult(False, Recursos.MensajesValidaciones.CorreoElectronico)
            Else
                Return ValidationResult.ValidResult
            End If
        End Function
    End Class
End Namespace